notebooks/Dynamic Double Machine Learning Examples.ipynb

```diff
@@ -0,0 +1,692 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define DGP parameters\n",
+    "np.random.seed(123)\n",
+    "n_panels = 400 # number of panels\n",
+    "n_periods = 3 # number of time periods in each panel\n",
+    "n_treatments = 1 # number of treatments in each period\n",
+    "n_x = 100 # number of features + controls\n",
+    "s_x = 10 # number of controls (endogeneous variables)\n",
+    "s_t = 10 # treatment support size"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Generate data\n",
+    "dgp = DynamicPanelDGP(n_periods, n_treatments, n_x).create_instance(\n",
+    "            s_x, random_seed=12345)\n",
+    "Y, T, X, W, groups = dgp.observational_data(n_panels, s_t=s_t, random_seed=12345)\n",
+    "true_effect = dgp.true_effect"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1.2 Train Estimator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "alpha_regs = [1e-4, 1e-3, 1e-2, 5e-2, .1, 1]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "est = DynamicDML(\n",
+    "    model_y=LassoCV(cv=3, alphas=alpha_regs, tol=1e-2), \n",
+    "    model_t=MultiTaskLassoCV(cv=3, alphas=alpha_regs, tol=1e-2), \n",
+    "    cv=3)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<econml.dml.dynamic_dml.DynamicDML at 0x1da18865cc8>"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "est.fit(Y, T, X=None, W=W, groups=groups, inference=\"auto\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Average effect of default policy: 2.26\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Average treatment effect of all periods on last period for unit treatments\n",
+    "print(f\"Average effect of default policy: {est.ate():0.2f}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Effect of target policy over baseline policy: 4.33\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Effect of target policy over baseline policy\n",
+    "# Must specify a treatment for each period\n",
+    "baseline_policy = np.zeros((1, n_periods))\n",
```

Comment:
Since the notebook can specify the treamtent dimension in the beginning it would be great if the whole notebook works when I set n_treatments > 1

Currently the code breaks

Reply:
Done